### PR TITLE
Add AKS check acr to Azure package

### PR DESCRIPTION
**Pull request by tbarnes94**
Added two methods to pkg/azure/acr.go:

1. ListRoleAssignments - mimics ListContainerRegistries.  This lists out all of the role assignments corresponding to the given subscriptionId and resourceGroup so we can check for the properties of an ACR pull role.
2. CheckACRPullAccess - uses 1) to determine whether or not the cluster has acr pull permissions.

### CODE REVIEW

File: pkg/azure/acr.go [hunk 1]
```diff
@@ -4,11 +4,31 @@ import (
...
 	"github.com/azure/spin-aks-plugin/pkg/logger"
 )
 
+type Role struct {
+	Name string
+	ID   string
+}
+
+const (
+	// tbarnes94: placeholder for now
```
Comment (OliverMKing) [inline pkg/azure/acr.go:18]: why is this a placeholder?

File: pkg/azure/acr.go [hunk 2]
```diff
@@ -4,11 +4,31 @@ import (
...
+	subscriptionId = "8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8"
```
Comment (imiller31) [inline pkg/azure/acr.go:19]: We shouldn't be commiting subscription ids at all? Should probably scrub this from commit history too

File: pkg/azure/acr.go [hunk 3]
```diff
@@ -53,3 +73,51 @@ func ListContainerRegistries(ctx context.Context, subscriptionId, resourceGroup
...
+	roles, err := ListRoleAssignments(ctx, subscriptionId, resourceGroup)
+	if err != nil {
+		return fmt.Errorf("listing role assignments: %w", err)
+	}
+	for _, role := range roles {
+		if (*role.Name == AcrPullRole.Name) && (*role.ID == AcrPullRole.ID) {
+			// tbarnes94: success case
+			return nil
+		}
+	}
```
Comment (OliverMKing) [inline pkg/azure/acr.go:87]: this alone does not tell us if a cluster can pull from an acr. We must check the scopes and ensure the principal id given to the is the kubelet identity.

This function must take a cluster as a parameter

Comment (OliverMKing) [reply pkg/azure/acr.go:87]: We want to check that a specific cluster has access to a specific acr

File: pkg/azure/acr.go (hunk 1 above)
Comment (tbarnes94) [reply pkg/azure/acr.go:18]: it's not

File: pkg/azure/acr.go (hunk 2 above)
Comment (tbarnes94) [reply pkg/azure/acr.go]: agree - will try squashing commits

**Review by davidgamero — APPROVED** [review]
lgtm

**Pull request merged**